proposals/0035-linalg-matrix.md

```diff
@@ -775,44 +794,54 @@ matrix object. When storing to `groupshared` memory, the matrix component data
 is converted to the target arithmetic or packed data type if the data types do
 not match.
 
-#### Matrix::Accumulate
+For the `Store` operations on `[RW]ByteAddressBuffers`, the `Stride` argument
+represents the row or column stride in bytes. For the `Store` operations on
+`groupshared` arrays, the `Stride` argument is the count of elements in the
+`groupshared` array.
+
+#### Matrix::InterlockedAccumulate
 
 ```c++
-template <MatrixScope ScopeLocal = Scope>
-typename hlsl::enable_if<Use == MatrixUse::Accumulator && ScopeLocal == Scope,
+
+// When Scope != Thread, the following overloads are available:
+template <MatrixUseEnum UseLocal = Use>
+typename hlsl::enable_if<Use == MatrixUse::Accumulator && UseLocal == Use,
                          void>::type
-Matrix::Accumulate(RWByteAddressBuffer Res, uint StartOffset, uint Stride,
-                   MatrixLayout Layout, uint Align = sizeof(ElementType));
+Matrix::InterlockedAccumulate(RWByteAddressBuffer Res, uint StartOffset,
+                              uint Stride, MatrixLayoutEnum Layout,
+                              uint Align = sizeof(ElementType));
 
-template <typename T, MatrixUse UseLocal = Use>
+template <typename T, MatrixUseEnum UseLocal = Use>
 typename hlsl::enable_if<hlsl::is_arithmetic<T>::value &&
-                             Use == MatrixUse::Accumulator &&
-                             Scope != MatrixScope::Thread && UseLocal == Use,
+                             Use == MatrixUse::Accumulator && UseLocal == Use,
                          void>::type
-Matrix::Accumulate(/*groupshared*/ T Arr[], uint StartIdx, uint Stride,
-                   MatrixLayout Layout);
-```
+Matrix::InterlockedAccumulate(/*groupshared*/ T Arr[], uint StartIdx,
+                              uint Stride, MatrixLayoutEnum Layout);
 
-The following table specifies the valid values for the `Layout` parameter
-given the `Accumulate` method type and matrix scope.  All other combinations
-are unsupported:
+// When Scope == Thread, the following overload is available:
+void Matrix::InterlockedAccumulate(RWByteAddressBuffer Res, uint StartOffset,
```

Comment:
is `Stride` missing from the param list here?

Reply:
Stride must be 0 for thread scope because the layout must be outer product optimal, so I removed the parameter.